src/internal/operators/takeLast.ts

```diff
@@ -39,18 +39,24 @@ import { MonoTypeOperatorFunction, TeardownLogic } from '../types';
  *
  * @throws {ArgumentOutOfRangeError} When using `takeLast(i)`, it delivers an
  * ArgumentOutOrRangeError to the Observer's `error` callback if `i < 0`.
+ * @throws {TypeError} If the count is not provided or is not a number.
  *
- * @param {number} count The maximum number of values to emit from the end of
+ * @param count The maximum number of values to emit from the end of
  * the sequence of values emitted by the source Observable.
- * @return {Observable<T>} An Observable that emits at most the last count
+ * @return An Observable that emits at most the last count
  * values emitted by the source Observable.
- * @method takeLast
- * @owner Observable
  */
 export function takeLast<T>(count: number): MonoTypeOperatorFunction<T> {
+  if (isNaN(count)) {
```

Comment:
I think we should use `Number.isNaN` here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN

Reply:
IIRC, that would only return true if it's *actually* `NaN`. People could pass in 'blah' there and it would be `false` if we used `Number.isNaN`.